app/controllers/users/registrations_controller.rb

```diff
@@ -0,0 +1,2 @@
+class Users::RegistrationsController < Devise::RegistrationsController
```

Comment:
k viết gì thì xóa đi e

Reply:
Nếu xóa file này thì sẽ mất Users::RegistrationsController và báo lỗi ạ, do mình có custom view và khai báo controller ở routes.rb

Reply:
Devise lưu password cũng giống như has_secure_password, có 2 trường password và password_confirmation, sử dụng gem bcrypt để encrypt trường password, lưu vào trường encrypted_password trong database ạ

Reply:
thuật toán mã hóa là gì e ?

Reply:
- Trước tiên thêm 1 string vào sau password : password =  password = "#{password}#{klass.pepper}"

- Dùng method cả Bcrypt để băm password sau khi thêm pepper: BCrypt::Password.create(password, cost: klass.stretches).to_s,

Reply:
thuật toán mã hóa MD5 e nhé, tìm hiểu thêm về thuật toán này đi e